benchmark/Usage.md

```diff
@@ -94,69 +100,90 @@ BenchmarkResults runAllBenchmarks(){
 It will now be compiled. The compiled version can be found inside the `benchmark` folder inside your build directory.
 
 # Using advanced benchmark features
+
 ## Metadata
+
 Setting metadata is handled by the `BenchmarkMetadata` class. The set metadata information will be included in the printed output of a compiled benchmark file and in the JSON file export.
 
 You can find instances of `BenchmarkMetadata` for your usage at 4 locations:
+
 - At `metadata_` of created `ResultEntry` objects, in order to give metadata information about the benchmark measurement.
+
 - At `metadata_` of created `ResultGroup` objects, in order to give metadata information about the group.
+
 - At `metadata_` of created `ResultTable` objects, in order to give metadata information about the table.
+
 - Writing a `getMetadata` function, like in the `BenchmarkInterface`, in order to give more general metadata information about your benchmark class. This is mostly, so that you don't have to constantly repeat metadata information, that are true for all the things you are measuring, in other places. For example, this would be a good place to give the name of an algorithm, if your whole benchmark class is about measuring the runtimes of one. Or you could give the time, at which those benchmark measurements were taken.
 
 ## Runtime configuration
+
 Passing values to your benchmark at runtime is possible and has its own system, split up into two phases.  
-Adding the configuration options and passing values to them.
+Defining the configuration options and passing values to them.
 
 ### Adding options
-Adding configuration options is done with the `addConfigurationOptions` function. More specifically, it's done by adding `BenchmarkConfigurationOption` objects to the passed instance of `BenchmarkConfiguration`, using the apt named `addConfigurationOption` function. These `BenchmarkConfigurationOption` objects are created using the `makeBenchmarkConfigurationOption` function.
 
-A `BenchmarkConfigurationOption` describes four things about a configuration option:
+Adding configuration options is done by adding configuration option to the protected member variable `manager_`, by using the function `ConfigManager::createConfigOption`. That is best done in the constructor of your class.
+
+In our system a configuration option is described by a handful of characteristics:
+
+1. The name of the option.
 
-1. What **type** of values it takes. The following types are available:
-  - Boolean.
-  - `std::string`.
-  - Integer.
-  - Floating point.
-  - A `std::vector` of the previous options.
+2. A description of the option.
 
-2. The name of the option.
+3. If it has a default value. If it hasn't, people will always have to provide their own value at run time.
 
-3. A description of the option.
+4. What **type** of values it takes. The following types are available:
+   
+   - Boolean.
+   - `std::string`.
+   - Integer.
+   - `size_t`
+   - Floating point.
```

Comment:
Don't mix the repsentations. They should all be exactly the types as they are called:
`bool` instead of `Boolean` etc.

Reply:
Done, see [cc5621e](https://github.com/ad-freiburg/qlever/pull/977/commits/cc5621ef5cf0d8a6d97e202ebfa90f7f16050a87).